File: packages/dotnet/src/Codec.Net/ITokenizer.cs
// SPDX-License-Identifier: MIT
namespace Codec;

/// <summary>
/// Common interface every tokenizer implementation satisfies.
/// <see cref="BPETokenizer"/>, <see cref="LongestMatchTokenizer"/>, and
/// any external/wasm adapter all implement this.
/// </summary>
public interface ITokenizer
{
    /// <summary>Identifier of the underlying vocabulary.</summary>
    string Id { get; }

    /// <summary>Encode a string to a sequence of token IDs.</summary>
    int[] Encode(string text);
}

File: packages/java/src/main/java/ai/codec/ITokenizer.java
// SPDX-License-Identifier: MIT
package ai.codec;

/**
 * Common interface every tokenizer implementation satisfies.
 * {@link BPETokenizer}, {@link LongestMatchTokenizer}, and any
 * external/wasm adapter all implement this.
 */
public interface ITokenizer {
    /** Identifier of the underlying vocabulary. */
    String getId();

    /** Encode a string to a sequence of token IDs. */
    int[] encode(String text);
}
